src/redisearch_api.c

```diff
@@ -394,13 +396,14 @@ typedef struct {
 
 static RS_ApiIter* handleIterCommon(IndexSpec* sp, QueryInput* input, char** error) {
   // here we only take the read lock and we will free it when the iterator will be freed
-  RWLOCK_ACQUIRE_READ();
+  RWLOCK_ACQUIRE_READ(sp);
 
   RedisSearchCtx sctx = SEARCH_CTX_STATIC(NULL, sp);
   RSSearchOptions options = {0};
   QueryError status = {0};
   RSSearchOptions_Init(&options);
   RS_ApiIter* it = rm_calloc(1, sizeof(*it));
+  it->sp = sp;
```

Comment:
What if you drop the index while there is an iterator running? Another reason to take the write lock on drop...